packages/flutter/lib/src/widgets/autocomplete.dart

```diff
@@ -0,0 +1,584 @@
+/// class User {
+///   const User({
+///     this.email,
+///     this.name,
+///   });
+///
+///   final String email;
+///   final String name;
+///
+///   // The displayStringForOption parameter can be used to get even more
+///   // control over the strings that represent the options objects.
+///   @override
+///   String toString() {
+///     return '$name, $email';
+///   }
+/// }
+///
+/// class AutocompleteCustomTypeExample extends StatelessWidget {
+///   AutocompleteCustomTypeExample({Key key});
+///
+///   final List<User> _kUserOptions = <User>[
+///     User(name: 'Alice', email: 'alice@example.com'),
+///     User(name: 'Bob', email: 'bob@example.com'),
+///     User(name: 'Charlie', email: 'charlie123@gmail.com'),
+///   ];
+///
+///   static String _displayStringForOption(User option) => option.name;
+///
+///   @override
+///   Widget build(BuildContext context) {
+///     return AutocompleteCore<User>(
+///       optionsBuilder: (TextEditingValue textEditingValue) {
+///         return _kUserOptions.where((User option) {
+///           return option.toString().contains(textEditingValue.text.toLowerCase());
+///         });
+///       },
+///       displayStringForOption: _displayStringForOption,
+///       fieldViewBuilder: (BuildContext context, TextEditingController textEditingController, VoidCallback onFieldSubmitted) {
+///         return TextFormField(
+///           controller: textEditingController,
+///           onFieldSubmitted: (String value) {
+///             onFieldSubmitted();
+///           },
+///         );
+///       },
+///       optionsViewBuilder: (BuildContext context, AutocompleteOnSelected<User> onSelected, Iterable<User> options) {
+///         return SizedBox(
+///           height: 200.0,
+///           child: Material(
+///             elevation: 4.0,
+///             child: ListView(
+///               padding: EdgeInsets.all(8.0),
+///               children: options.map((User option) => GestureDetector(
+///                 onTap: () {
+///                   onSelected(option);
+///                 },
+///                 child: ListTile(
+///                   title: Text(_displayStringForOption(option)),
+///                 ),
+///               )).toList(),
+///             ),
+///           ),
+///         );
+///       },
+///     );
+///   }
+/// }
+/// ```
+/// {@end-tool}
+///
+/// {@tool dartpad --template=freeform}
+/// This example shows the use of AutocompleteCore in a form.
+///
+/// ```dart imports
+/// import 'package:flutter/widgets.dart';
+/// import 'package:flutter/material.dart';
+/// ```
+///
+/// ```dart
+/// class AutocompleteFormExamplePage extends StatefulWidget {
+///   AutocompleteFormExamplePage({Key key}) : super(key: key);
+///
+///   @override
+///   AutocompleteFormExample createState() => AutocompleteFormExample();
+/// }
+///
+/// class AutocompleteFormExample extends State<AutocompleteFormExamplePage> {
```

Comment:
I wish large examples like this were references to repos that contained tests. 

CC @gspencergoog

Reply:
Yes, that would be nice. I believe this is already filed (by you) as https://github.com/flutter/flutter/issues/56163

Reply:
Agreed!  In fact, I'm already maintaining these examples (and a few others) in an [external repo](https://github.com/justinmc/flutter-autocomplete-examples).  It's much easier for me to run them there and then paste the relevant code here.  Otherwise it was too hard to make sure they still work after changes.